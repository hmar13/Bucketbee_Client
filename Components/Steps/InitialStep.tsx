import React, { useContext, Dispatch, SetStateAction } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { AuthContext } from '../../Operations/Store';
import * as Animatable from 'react-native-animatable';
import { Button } from 'react-native-paper';
import theme from '../../styles/theme.style';


type SetCountType = Dispatch<SetStateAction<number>>

interface Props {
  setCount: SetCountType
}

const InitialStep: React.FC<Props> = ({ setCount }) => {
  const { dispatch } = useContext(AuthContext);

  const handleConfirm = () => {
    setCount((val: number) => val + 1);
  };

  const handleSkip = () => {
    dispatch({ type: 'ENDFORM' });
  };

  return (
    <View style={styles.container}>
      <Animatable.View animation="fadeInUpBig" style={styles.wrapper}>
        <Text style={styles.text}>Welcome to Bucketbee!</Text>
        <Text style={styles.subtext}>Tell me a bit about yourself?</Text>
        <View
          style={{
            width: '100%',
            flexDirection: 'row',
          }}
        >
          <TouchableOpacity activeOpacity={0.7} onPress={handleSkip}>
            <Button
              style={styles.button}
              labelStyle={{ color: theme.PRIMARY_COLOR }}
            >
              MAYBE LATER
            </Button>
          </TouchableOpacity>
          <TouchableOpacity activeOpacity={0.7} onPress={handleConfirm}>
            <Button
              style={styles.button}
              labelStyle={{ color: theme.PRIMARY_COLOR }}
            >
              SURE!
            </Button>
          </TouchableOpacity>
        </View>
      </Animatable.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.PRIMARY_COLOR,
  },
  wrapper: {
    height: 200,
    width: 300,
    justifyContent: 'center',
  },
  button: {
    backgroundColor: theme.PRIMARY_COLOR_XLITE,
    marginVertical: 50,
    marginRight: 40,
    paddingVertical: 8,
    color: theme.PRIMARY_COLOR,
    minWidth: 100,
    justifyContent: 'center',
  },
  text: {
    color: theme.PRIMARY_COLOR_XLITE,
    fontSize: 24,
    fontFamily: 'Poppins_400Regular',
    paddingBottom: 8,
  },
  subtext: {
    color: theme.PRIMARY_COLOR_XLITE,
    fontSize: 22,
    fontFamily: 'Lato_300Light',
  },
});

export default InitialStep;
