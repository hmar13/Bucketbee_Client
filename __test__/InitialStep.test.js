import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Button } from 'react-native-paper';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import InitialStep from '../Components/Steps/InitialStep';

describe('<InitialStep />', () => {
  const setCount = jest.fn();
  const wrapper = shallow(<InitialStep
    setCount={setCount}
  />);

  test('Renders correctly', () => {
    const tree = renderer.create(<InitialStep />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render the second Button Text', () => {
    expect(wrapper.find(Button).at(1).props().children).toEqual('SURE!')
  });

  test('Should call setCount on Touchable press', () => {
    wrapper.find(TouchableOpacity).at(1).simulate('press');
    expect(setCount).toHaveBeenCalled();
  })

});
