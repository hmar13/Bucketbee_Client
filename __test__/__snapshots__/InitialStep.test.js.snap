// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InitialStep /> Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d525f",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "height": 200,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9affdc",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 24,
          "paddingBottom": 8,
        }
      }
    >
      Welcome to Bucketbee!
    </Text>
    <Text
      style={
        Object {
          "color": "#9affdc",
          "fontFamily": "Lato_300Light",
          "fontSize": 22,
        }
      }
    >
      Tell me a bit about yourself?
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#9affdc",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "color": "#0d525f",
              "elevation": 0,
              "justifyContent": "center",
              "marginRight": 40,
              "marginVertical": 50,
              "minWidth": 100,
              "paddingVertical": 8,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "color": "#0d525f",
                      },
                    ],
                  ]
                }
              >
                MAYBE LATER
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#9affdc",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "color": "#0d525f",
              "elevation": 0,
              "justifyContent": "center",
              "marginRight": 40,
              "marginVertical": 50,
              "minWidth": 100,
              "paddingVertical": 8,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "color": "#0d525f",
                      },
                    ],
                  ]
                }
              >
                SURE!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
